src/vdom.rs

```diff
@@ -195,29 +197,67 @@ impl<Ms, Mdl, ElC: View<Ms> + 'static, GMs: 'static> App<Ms, Mdl, ElC, GMs> {
         }
     }
 
+    /// Bootstrap the dom with the vdom by taking over all children of the mount point and
+    /// replacing them with the vdom if requested. Will otherwise ignore the original children of
+    /// the mount point.
     fn bootstrap_vdom(&self) -> El<Ms> {
+        let mount_type = self.cfg.mount_type.borrow().unwrap_or(MountType::Append);
         // "new" name is for consistency with `update` function.
         // this section parent is a placeholder, so we can iterate over children
         // in a way consistent with patching code.
         let mut new = El::empty(dom_types::Tag::Placeholder);
 
+        // Map the DOM's elements onto the virtual DOM if requested to takeover.
+        if mount_type == MountType::Takeover {
+            // Construct a vdom from the root element. Subsequently strip the workspace so that we
+            // can recreate it later - this is a kind of simple way to avoid missing nodes (but
+            // not entirely correct).
+            // TODO: 1) Optimize by utilizing a patching strategy instead of recreating the workspace
+            // TODO: nodes.
+            // TODO: 2) Watch out for elements that should not be recreated, such as script nodes,
+            // TODO: and other, similar things. For now, leave the warning in the builder's
+            // TODO: documentation.
+            let mut dom_nodes: El<Ms> = (&self.cfg.mount_point).into();
+            dom_nodes.strip_ws_nodes_from_self_and_children();
+
+            // Replace the root dom with a placeholder tag and move the children from the root element
+            // to the newly created root. Uses `Placeholder` to mimic update logic.
+            new.children = dom_nodes.children;
+        }
+
         self.setup_window_listeners();
         patch::setup_input_listeners(&mut new);
         patch::attach_listeners(&mut new, &self.mailbox());
 
-        websys_bridge::assign_ws_nodes_to_el(&util::document(), &mut new);
+        // Recreate the needed nodes. Only do this if requested to takeover the mount point since
+        // it should only be needed here.
+        if mount_type == MountType::Takeover {
+            // TODO: Refer the Optimization TODO.
```

Comment:
So.. what's the TODO?  Can I remove this comment once the optimization issue is resolved?

Reply:
Yes. It's the TODO referenced in the other comment.